test/core/test/web-worker-node.test.ts

```diff
@@ -203,6 +203,7 @@ it('self injected into worker and its deps should be equal', async () => {
   await sleep(0)
   expect(await testSelfWorker(new MySelfWorker())).toBeTruthy()
 
+  vi.resetModules()
```

Comment:
The test should work without clearing modules, that's the whole point of it, not?

Reply:
I thought this test actually contained "two tests", as in first the `MySelfWorker` part and then `Worker` part. 

If this test should pass without module reset, we cannot move the module invalidation in `InlineWorker` to happen before `executeFile`. So cannot use `ModuleCache` for holding the `startOffset`.

Reply:
This is a single test, the worker is the same, just initiated differently. 

> So cannot use `ModuleCache` for holding the `startOffset`.

Seems like we need to go to the drawing board then